java/server/test/org/openqa/selenium/grid/graphql/GraphqlHandlerTest.java

```diff
@@ -120,11 +139,260 @@ public boolean test(Capabilities capabilities) {
       .isEqualTo(Map.of("data", Map.of("grid", Map.of("nodes", List.of(Map.of("uri", nodeUri))))));
   }
 
+  @Test
+  public void shouldBeAbleToGetSessionInfo() throws URISyntaxException {
+    String nodeUrl = "http://localhost:5556";
+    URI nodeUri = new URI(nodeUrl);
+
+    Node node = LocalNode.builder(tracer, events, nodeUri, publicUri, registrationSecret)
+        .add(caps, new TestSessionFactory((id, caps) -> new org.openqa.selenium.grid.data.Session(
+            id,
+            nodeUri,
+            stereotype,
+            caps,
+            Instant.now()))).build();
+
+    distributor.add(node);
+    Session session = distributor.newSession(createRequest(payload)).getSession();
+
+    assertThat(session).isNotNull();
+    String sessionId = session.getId().toString();
+
+    Set<Slot> slots = distributor.getStatus().getNodes().stream().findFirst().get().getSlots();
+
+    Slot slot = slots.stream().findFirst().get();
+
+    org.openqa.selenium.grid.graphql.Session graphqlSession =
+        new org.openqa.selenium.grid.graphql.Session(sessionId,
+                                                     session.getCapabilities(),
+                                                     session.getStartTime(),
+                                                     session.getUri(),
+                                                     node.getId().toString(),
+                                                     node.getUri(),
+                                                     slot);
+    String query = "{ session (id: \"" + sessionId + "\") { id, capabilities, startTime, uri } }";
+
+    GraphqlHandler handler = new GraphqlHandler(distributor, publicUri);
+    Map<String, Object> result = executeQuery(handler, query);
+
+    assertThat(result)
+        .describedAs(result.toString())
+        .isEqualTo(Map.of("data", Map.of("session",
```

Comment:
Nit: try not to have the indent hanging so far to the right. In this case, I think putting a newline before `data` would preserve intent and make the code a little neater to read.